src/main/java/net/neoforged/neoforge/common/util/NeoForgeExtraCodecs.java

```diff
@@ -143,4 +165,46 @@ public String toString() {
             return "Alternative[" + codec + ", " + alternative + "]";
         }
     }
+
+    private static class AlternativeMapCodec<TYPE> extends MapCodec<TYPE> {
```

Comment:
That would be `T` following the usual conventions. :P (and I suppose `T1` for the extra generic in the methods below)